Update Colors And Dark Mode Toggle to Improve A11Y

kinson opened the pull request:
# Overview

Addresses (but does not close) #35

* Updated the colors used on the website to increase the contrast between text and background to meet WCAG 2 AA contrast thresholds.
* Updated the dark mode toggle to not use a label as a wrapper element. Further improved accessibility of this control by labeling the input itself.
* Added page titles to controllers that were missing them
* Converted some `p` and `span` tags to header tags to better semantically describe the content

doomspork commented:
@kinson this is really amazing and beyond my scope of knowledge, do you have links that make some of this easy to ingest?

kinson commented:
@doomspork sure do!

The color contrast thresholds are new to me as well. To get some background on what/why/how, I skimmed through [this page](https://webaim.org/articles/contrast/), but I mostly relied on this [contrast checker](https://webaim.org/resources/contrastchecker/) tool to take the offending elements and determine a better (higher contrast) color for them.

The dark mode toggle turned out to violate a html standard by having the whole control wrapped in a `<label>` tag. Admittedly, I used this strategy in the first place to get around using an on click handler for the `dark-mode-toggle` checkbox input. To fix that I used a div instead and added an on click handler to it (hence the updates to the dark mode js file).

I also added the `aria-label` property to the checkbox input element since the control itself describes the functionality so adding a label would not make sense. That _should_ resolve the [`Form <input> elements must have labels`](https://rocketvalidator.com/accessibility-validation/label/wcag%3A-1.3.1%2C-4.1.2-each-form-element-must-have-a-programmatically-associated-label-element) issue from Rocket Validator.

I am curious to see how this affects the report, I am pretty sure this will not resolve all of the issues but it will hopefully make a dent in the number of issues and make the site more usable/accessible 🎉

doomspork reviewed (approved):
Pulled this down and things look good 👍
